pkg/reconciler/common/events.go

```diff
@@ -18,6 +18,15 @@ package common
 
 // Reasons for API Events
 const (
+	// ReasonRBACCreate indicates that an RBAC object was successfully created.
+	ReasonRBACCreate = "CreateRBAC"
+	// ReasonRBACUpdate indicates that an RBAC object was successfully updated.
+	ReasonRBACUpdate = "UpdateRBAC"
+	// ReasonFailedRBACCreate indicates that the creation of an RBAC object failed.
+	ReasonFailedRBACCreate = "FailedRBACCreate"
+	// ReasonFailedRBACUpdate indicates that the update of an RBAC object failed.
+	ReasonFailedRBACUpdate = "FailedRBACUpdate"
+
```

Comment:
Are we also going to handle the ServiceAccount creation/updates as well?

Reply:
Not sure what you meant there, but if you're asking whether we also reconcile the Service Account _after_ its creation, the answer is yes (reconcile.go -> `syncServiceAccount()`)

Reply:
I think I got it.

We need to delegate the management of Service Accounts to some controller because those should only exist _if_ at least one source of the corresponding type exists. The manual way would be to create one service account for each possible source/target type in every Kubernetes namespace, and that is not very efficient in terms of housekeeping.

We do keep the RBAC _Role_ cluster-scoped though, so the management of the actual permissions remains centralized. Only the fact of binding those permissions to an actual application is per-namespace.

Reply:
Given the service account must exist prior to binding, I was expecting the event notification for it's creation to also be generated. However, if you would prefer to keep it coupled, then it should be okay as well.

Reply:
It is generated. What am I missing?

Reply:
More like reporting `FailedServiceAccountCreate` event when failing to create the service account instead of `FailedRBACCreate`.

Reply:
"of an RBAC object" is to be understood as "of any RBAC" object here. I'm setting the same Reason for both Service Accounts and Role Bindings, but the value of Message differs and contains the detail of the operation.

Not sure if it brings any value to create many Reason constants. I was even thinking "Create object" / "UpdateObject" would be enough and could be used for any object, but didn't want to bikeshed.

Reply:
Btw, we also don't have different event reasons depending on whether the adapter is a Deployment or a Ksvc, so I'm simply remaining consistent.

Reply:
Oh no I get it. Forgive me, this is more for gaining understanding and not a critique.